api/Trace/TextMapPropagator.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Trace;
+
+use OpenTelemetry\Context\Context;
+
+/**
+ * @see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md#textmap-propagator
+ */
+interface TextMapPropagator
+{
+    /**
+     * Returns list of fields that will be used by this propagator.
+     *
+     * @see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md#fields
+     *
+     * @return list<string>
+     */
+    public static function fields() : array;
+
+    /**
+     * Injects specific values from the provided {@see Context} into the provided carrier
+     * via an {@see PropagationSetter}.
+     *
+     * @see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md#textmap-inject
+     *
+     * @param mixed $carrier
+     */
+    public static function inject(&$carrier, Context $context = null, PropagationSetter $setter = null): void;
```

Comment:
I took some inspiration from the Ruby implementation where it made the context and setter optional to fall back on current context and the default setter (`PropagationMap` in our case).

Just makes it a bit easier to use, but happy to not do that if we'd rather be more explicit.

Reply:
Agreed that this makes things a bit easier to use; I'm all for it.